src/java/org/apache/cassandra/index/sai/plan/Expression.java

```diff
@@ -208,6 +211,27 @@ public Expression add(Operator op, ByteBuffer value)
                 else
                     lower = new Bound(value, indexTermType, lowerInclusive);
                 break;
+
+            case BETWEEN:
+                ListType<?> type = ListType.getInstance(indexTermType.columnMetadata().type, false);
+                List<? extends ByteBuffer> buffers = type.unpack(value);
+
+                operator = IndexOperator.RANGE;
+                this.upperInclusive = true;
+                this.lowerInclusive = true;
+
+                if (indexTermType.isReversed())
+                {
+                    upper = new Bound(buffers.get(0), indexTermType, true);
+                    lower = new Bound(buffers.get(1), indexTermType, true);
+                }
+                else
+                {
+                    lower = new Bound(buffers.get(0), indexTermType, true);
+                    upper = new Bound(buffers.get(1), indexTermType, true);
+                }
```

Comment:
An alternate way to express this, if you think it reads better (not binding by any means) is...

```
int lowerIndex = indexTermType.isReversed() ? 1 : 0;
int upperIndex = indexTermType.isReversed() ? 0 : 1;
lower = new Bound(buffers.get(lowerIndex), indexTermType, true);
upper = new Bound(buffers.get(upperIndex), indexTermType, true);
```